packages/commonwealth/client/scripts/helpers/ContractHelpers/Contest.ts

```diff
@@ -0,0 +1,125 @@
+import { AppError } from '@hicommonwealth/core';
+import { ContestAbi } from './Abi/ContestAbi';
+import ContractBase from './ContractBase';
+import NamespaceFactory from './NamespaceFactory';
+
+class Contest extends ContractBase {
+  namespaceFactoryAddress: string;
+  namespaceFactory: NamespaceFactory;
+
+  constructor(contractAddress: string, factoryAddress: string, rpc: string) {
+    super(contractAddress, ContestAbi, rpc);
+    this.namespaceFactoryAddress = factoryAddress;
+  }
+
+  async initialize(withWallet: boolean = false): Promise<void> {
+    await super.initialize(withWallet);
+    this.namespaceFactory = new NamespaceFactory(
+      this.namespaceFactoryAddress,
+      this.rpc,
+    );
+    await this.namespaceFactory.initialize();
+  }
+
+  /**
+   * deploys a new recurring contest and configures the feeManager to divert feeShare of
+   * stake fees to new contest
+   * @param namespaceName namespace name to configure for
+   * @param contestInterval the recurrng contest interval in ms
+   * @param winnerShares the percent of each respective winning places prize(ie 5 = 5%)
+   * @param stakeId the id of the community stake token, defaults to 2
+   * @param prizeShare the % amount of the prize pool to payout each interval(ie 5 = 5%)
+   * @param voterShare the % amount of prize pool claimable by voters each week(ie 5 = 5%)
+   * @param feeShare the % amount of community stake fees to divert to contest(ie 5 = 5%)
+   * @param weight the weight of each stake balance, defaults to 1
```

Comment:
can you remind be how do I access this data on the FE?

Reply:
I don't think its currently accessed on the front end but rather calculated backend. I think for now we can use the 1 default until requirements for dynamic weights are provided by product. 

When that time comes the weight is stored on the CommunityStake.vote_weight model in the DB. But it currently always default to 1 on backend too.

Note: this is our own off-chain data/concept